### PR TITLE
Hide/show widget [Core] [Cocoa]

### PR DESCRIPTION
**Pull request by Dayof**
Resolves Issue #181

### CODE REVIEW

**Comment by phildini** [comment]
Hey there! Would you consider this ready for review, or are you still working on it?

**Comment by Dayof** [comment]
I think that can be reviewed. Later, this feature can be update with the addition of the constraints update.

File: src/core/toga/interface/widgets/base.py [hunk 1]
```diff
@@ -331,3 +330,21 @@ def set_font(self, font):
         :type  font: :class:`toga.Font`
         """
         self._set_font(font)
+
+    def hide(self, widget):
```
Comment (freakboy3742) [inline src/core/toga/interface/widgets/base.py]: Why does this need to take widget as an argument? Isn't `self` the widget that needs to be hidden?

File: src/cocoa/toga_cocoa/widgets/base.py [hunk 2]
```diff
@@ -14,6 +14,18 @@ def _set_container(self, container):
             self._constraints._container = container
         self.rehint()
 
+    def _set_hidden(self, status, children):
+        for view in self._container._impl.subviews:
```
Comment (freakboy3742) [inline src/cocoa/toga_cocoa/widgets/base.py]: Iterating over the list of children should be something common to *all* containers; could this be moved up to the interface layer?

File: src/cocoa/toga_cocoa/widgets/base.py [hunk 3]
```diff
@@ -14,6 +14,18 @@ def _set_container(self, container):
...
+            [view.setHidden_(status) for c in children if c._impl == view]
```
Comment (freakboy3742) [inline src/cocoa/toga_cocoa/widgets/base.py]: This will remove the child remove the child from the view; but it won't modify any of the CSS geometry.

Comment (phildini) [inline src/cocoa/toga_cocoa/widgets/base.py]: I'm also slightly confused why this is a list comprehension... using a list comprehension just for the side-effects seems odd to me.

While I'm here, where is `setHidden_` defined? I couldn't find it in a grep of the codebase?

**Comment by phildini** [comment]
It would be really thrilling to see anything that looks like a test or example code for this.

File: src/core/toga/interface/widgets/base.py (hunk 1 above)
Comment (Dayof) [inline src/core/toga/interface/widgets/base.py]: The idea is to use like this : self.outer_box.hide(self.label_box), but yeah I can update to self.label_box.hide().

File: src/cocoa/toga_cocoa/widgets/base.py (hunk 2 above)
Comment (Dayof) [inline src/cocoa/toga_cocoa/widgets/base.py]: Ok.

File: src/cocoa/toga_cocoa/widgets/base.py (hunk 3 above)
Comment (Dayof) [inline src/cocoa/toga_cocoa/widgets/base.py]: Yes, is not updating the super view geometry yet, I tryed to use one of the recommendation here "https://stackoverflow.com/questions/19561269/autolayout-with-hidden-uiviews" but is a problem to look more deeply.

**Comment by Dayof** [comment]
I'll commit the new demands and post a code as an example. Thanks for the review @phildini  and @freakboy3742  !

**Comment by freakboy3742** [comment]
@phildini The work @Ocupe is doing for the GSoC is aimed at making the Toga interface testable; hopefully we'll be able to add tests in the very near future.

**Comment by Dayof** [comment]
@phildini the example code requested : https://gist.github.com/Dayof/b47a18325d7af75708d74fd0c569f0a2

**Comment by Dayof** [comment]
@freakboy3742 @phildini @eliasdorneles does this PR still with some reviews to do?

**Pull request merged**